.vscode/settings.json

```diff
@@ -6,5 +6,25 @@
     },
     "search.exclude": {
         "out": true // set this to false to include "out" folder in search results
+    },
+    "eslint.autoFixOnSave":  true,
+    "eslint.validate":  [
+    "javascript",
+    "javascriptreact",
+    {"language":  "typescript",  "autoFix":  true  },
+    {"language":  "typescriptreact",  "autoFix":  true  }
+    ],
+    "editor.formatOnSave":  true,
+    "[javascript]":  {
+    "editor.formatOnSave":  false,
+    },
+    "[javascriptreact]":  {
+    "editor.formatOnSave":  false,
+    },
+    "[typescript]":  {
+    "editor.formatOnSave":  false,
+    },
+    "[typescriptreact]":  {
+    "editor.formatOnSave":  false,
     }
```

Comment:
why not format on save for TypeScript

Reply:
"formatOnSave" is conflict with "autoFixOnSave", so we should set autoFixOnSave" as true and "formatOnSave" as false. See https://github.com/prettier/prettier-vscode/issues/315